fl4health/server/nnunet_server.py

```diff
@@ -39,12 +56,77 @@ def new_fn(*args: Any, **kwargs: Any) -> Any:
     return new_fn
 
 
-class NnunetServer(FlServerWithInitializer):
-    """
-    A Basic FlServer with added functionality to ask a client to initialize
-    the global nnunet plans if one was not provided in the config. Intended
-    for use with NnUNetClient
-    """
+class NnunetServer(FlServerWithInitializer, FlServerWithCheckpointing):
+    def __init__(
+        self,
+        client_manager: ClientManager,
+        parameter_exchanger: ParameterExchanger,
+        model: Optional[nn.Module] = None,
+        wandb_reporter: Optional[ServerWandBReporter] = None,
+        strategy: Optional[Strategy] = None,
+        checkpointer: Optional[Union[TorchCheckpointer, Sequence[TorchCheckpointer]]] = None,
+        metrics_reporter: Optional[MetricsReporter] = None,
+        intermediate_server_state_dir: Optional[Path] = None,
+        server_name: Optional[str] = None,
+    ) -> None:
+        FlServerWithCheckpointing.__init__(
+            self,
+            client_manager=client_manager,
+            model=model,
+            parameter_exchanger=parameter_exchanger,
+            wandb_reporter=wandb_reporter,
+            strategy=strategy,
+            checkpointer=checkpointer,
+            metrics_reporter=metrics_reporter,
+            intermediate_server_state_dir=intermediate_server_state_dir,
+            server_name=server_name,
+        )
+        """
+        A Basic FlServer with added functionality to ask a client to initialize
+        the global nnunet plans if one was not provided in the config. Intended
+        for use with NnUNetClient
+        """
+        self.initialized = False
+
+        self.nnunet_plans_bytes: bytes
+        self.num_input_channels: int
+        self.num_segmentation_heads: int
+        self.enable_deep_supervision: bool
+        self.nnunet_config: NnunetConfig
+
+    def initialize_server_model(self) -> None:
+        # Ensure required attributes are set
+        assert (
+            self.nnunet_plans_bytes is not None
+            and self.num_input_channels is not None
+            and self.num_segmentation_heads is not None
+            and self.enable_deep_supervision is not None
+            and self.nnunet_config is not None
+        )
+
+        plans = pickle.loads(self.nnunet_plans_bytes)
+        plans_manager = PlansManager(plans)
+        configuration_manager = plans_manager.get_configuration(self.nnunet_config.value)
+        model = nnUNetTrainer.build_network_architecture(
+            configuration_manager.network_arch_class_name,
+            configuration_manager.network_arch_init_kwargs,
+            configuration_manager.network_arch_init_kwargs_req_import,
+            self.num_input_channels,
```

Comment:
I do like the idea of passing the dataset.json through the get_properties. I may be interpreting your second suggestion incorrectly but  assuming the server and client are on the same file system may not be the best assumption going forward when we try to actually do this in a distributed manner. So in that way I prefer your first suggestion but feel free to correct me if I misinterpreted the suggestion

Reply:
You don't need to assume they are on the same file system. You could pass the actual dict that is stored in the dataset.json. Each client in theory should have their own copy of this on their own system since it is already required for the client setup. But as I mentioned in my next comment that you responded too, this might be an unnecessary optimization

Reply:
Ahh I see what you mean, sorry about the confusion